core/lib/src/config/config.rs

```diff
@@ -89,8 +89,19 @@ pub struct Config {
     /// that the value must syntactically be a valid HTTP header name.
     ///
     /// **(default: `"X-Real-IP"`)**
-    #[serde(deserialize_with = "crate::config::ip_header::deserialize")]
+    #[serde(deserialize_with = "crate::config::http_header::deserialize")]
     pub ip_header: Option<Uncased<'static>>,
+    /// The name of a header, whose value is typically set by an intermediary
+    /// server or proxy, which contains the protocol used by the connecting
+    /// client. Used by [`Request::proxy_proto()`] and
+    /// [`Request::proxy_secure()`].
```

Comment:
We need some more guidance here. It's _really_ important that this header is trusted. Otherwise, we might assume we're in a secure context when we're not. For this PR (cookies being marked as secure), we just create an application that doesn't. But that might change in the future.

We could also say that this should probably be `X-Forwarded-Proto` and point to docs for it.

Reply:
I updated to docs to add some more guidance and a link to the docs, hope its better now.